src/book/02-system/01-foundations/02-build-system.mdx

```diff
@@ -344,6 +345,44 @@ The `nuc7i7bnh` target should be used for building and running code on the robot
 
 The `target` command will download (or, if needed, build) the Docker image for the specified target and then mark that target as active.
 
+##### Tests
+
+The `tests` command allows you to run built unit tests. It will automatically run tests in parallel, and dump a timestamped log file to the project directory.
+Currently, we use `CTest` which is included as part of `CMake`.
+
+In order to run unit tests, they must first be built, this can be done with
+
+```bash
+./b configure -- -DBUILD_TESTS=ON
+./b build
+```
+
+The `tests` command provides the following subcommands
+
+| Subcommand | Description                 |
+| ---------- | --------------------------- |
+| list       | List all built tests        |
+| run        | Run all or individual tests |
+
+If you would like to run a individual test (or group of tests), you can pass the `run` subcommand a string and it will only run tests that match the given string.
+Alternatively, `run` will run all built tests if no string is passed.
+
+For example, if you would like to run all tests with the string 'vision' in their name, you would do
+
+```bash
+./b tests run vision
+```
+
+The `run` subcommand also accepts the following flags
+
+| Flag                               | Description                                               |
+| ---------------------------------- | --------------------------------------------------------- |
+| -V or --verbose                    | Enable verbose output from tests                          |
+| -VV or --extra-verbose             | Enable more verbose output from tests                     |
+| -Q or --quiet                      | Make ctest not print to stdout                            |
+| -j NUM_JOBS or --parallel NUM_JOBS | Run the tests in parallel using the given number of jobs. |
+| --debug                            | Displaying more verbose internals of CTest                |
```

Comment:
```suggestion
| `--debug`                            | Display verbose output from CTest internals                |
```

Reply:
These flag descriptions are taken directly from [ctest docs](https://cmake.org/cmake/help/latest/manual/ctest.1.html#options), and are what is displayed when you run `./b tests run -h`, so for consistancy I think the NUbook docs should match.

Reply:
Ah I see.

I've suggested the change because this line [isn't parallel](https://www.myenglishpages.com/site_php_files/grammar-lesson-parallelism.php) with the others. They all have verbs in the present tense but the last one is present progressive. E.g. "enable", "make", "run", _"displaying"_ vs "enable", "make", "run", _"display"_.

I think we can deviate slightly here for the correction, but we don't have to. Go with what you think is best.

Reply:
Ah ok, Yeah makes sense. I do think the suggestion is much better, but just to make it consistent with the `./b tests run -h` I will leave it as is. We can easily change it at a later date, but both should be changed.